nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-restore-content-from-coldstorage-button.html

```diff
@@ -0,0 +1,117 @@
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing permissions and limitations under the License.
+-->
+
+<!--
+`nuxeo-move-content-to-coldstorage-button`
+@group Nuxeo UI
+@element nuxeo-move-content-to-coldstorage-button 
+-->
+
+<dom-module id="nuxeo-restore-content-from-coldstorage-button">
+  <template>
+    <style include="nuxeo-action-button-styles nuxeo-styles"></style>
+    <nuxeo-operation id="opRestore" op="Document.RestoreFromColdStorage" input="[[document.uid]]"></nuxeo-operation>
+    <nuxeo-connection id="nxcon" user="{{currentUser}}"></nuxeo-connection>
+
+    <dom-if if="[[_isAvailable(document, currentUser)]]">
+      <template>
+        <div class="action" on-click="_toggleDialog">
+          <paper-icon-button icon="nuxeo:coldstorage" noink></paper-icon-button>
+          <span class="label">[[i18n('documentContentView.restoreFromColdStorage.label')]]</span>
+          <nuxeo-tooltip>[[i18n('documentContentView.restoreFromColdStorage.tooltip')]]</nuxeo-tooltip>
+        </div>
+      </template>
+    </dom-if>
+
+    <nuxeo-dialog id="contentToRestoreDialog" with-backdrop>
+      <h2>[[i18n('documentContentView.restoreFromColdStorage.ask.confirmation')]]</h2>
+      <p>[[i18n('documentContentView.restoreFromColdStorage.description')]]</p>
+
+      <div class="buttons">
+        <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+          [[i18n('documentContentView.restoreFromColdStorage.cancel')]]
+        </paper-button>
+        <paper-button id="confirm" name="confirm" class="primary" noink dialog-confirm on-click="_toggle">
+          [[i18n('documentContentView.restoreFromColdStorage.confirm')]]
+        </paper-button>
+      </div>
+    </nuxeo-dialog>
+  </template>
+  <script>
+    class RestoreFromColdStorage extends Polymer.mixinBehaviors(
+      [Nuxeo.FiltersBehavior, Nuxeo.FormatBehavior],
+      Nuxeo.Element,
+    ) {
+      static get is() {
+        return 'nuxeo-restore-content-from-coldstorage-button';
+      }
+
+      static get properties() {
+        return {
+          /**
+           * Input document.
+           */
+          document: Object,
+          /**
+           * Current user.
+           */
+          currentUser: Object,
+        };
+      }
+
+      _toggleDialog() {
+        this.$.contentToRestoreDialog.toggle();
+      }
+
+      _hasAdministrationPermissions(user) {
+        return user && (user.isAdministrator || this.isMember(user, 'powerusers'));
+      }
+
+      _isAvailable(document, currentUser) {
+        return (
+          this.hasFacet(document, 'ColdStorage')
+      && !document.properties['coldstorage:toBeRestored']
+      && (this._hasAdministrationPermissions(currentUser) || this.hasPermission(document, 'WriteColdStorage'))
+      && this.hasContent(document)
+        );
+      }
```

Comment:
I think it will be worth laster asap :) to see if it's better to extract a new behaviour. As the same rule is duplicated at least in two places: Move, Restore and Retrieve I am not sure. Could create a ticket to do it.
thx

Reply:
Good point! Will create a new ticket for it.

Reply:
Thanks

Reply:
Thanks